### PR TITLE
update go to 1.17 in go.mod

### PR DESCRIPTION
**Pull request by rockmenjack**
Signed-off-by: rockmenjack <rockmen1@gmail.com>

/kind dependency-change

#### What this PR does / why we need it:

golang 1.15 can no longer compile cri-o because https://github.com/kubernetes/client-go/commit/ef1d5d2691a6e54a5524eda7db07a918f806fbb2 has introduced `io/fs` which comes with golang 1.17

```
# make binaries
mkdir -p "/root/cri-o/_output/src/github.com/cri-o"
ln -s "/root/cri-o" "/root/cri-o/_output/src/github.com/cri-o/cri-o"
touch "/root/cri-o/_output/.gopathok"
go build -trimpath  -ldflags '-s -w -X github.com/cri-o/cri-o/internal/pkg/criocli.DefaultsPath="" -X github.com/cri-o/cri-o/internal/version.buildDate='2021-11-24T03:40:58Z' -X github.com/cri-o/cri-o/internal/version.gitCommit=2811bd6afea913deb176acd667b904978c273e81 -X github.com/cri-o/cri-o/internal/version.gitTreeState=clean ' -tags "containers_image_ostree_stub  exclude_graphdriver_btrfs btrfs_noversion exclude_graphdriver_devicemapper libdm_no_deferred_remove    " -o bin/crio github.com/cri-o/cri-o/cmd/crio
vendor/k8s.io/client-go/plugin/pkg/client/auth/exec/metrics.go:21:2: cannot find package "." in:
        /root/cri-o/vendor/io/fs
make: *** [Makefile:172: bin/crio] Error 1
```

#### Which issue(s) this PR fixes:

this PR just update `go.mod` to reflect dependency on golang 1.17+

```release-note
none
```

### CODE REVIEW

**Comment by haircommander** [comment]
/ok-to-test
/approve

LGTM, thank you @rockmenjack

File: go.mod [hunk 1]
```diff
@@ -1,4 +1,4 @@
-go 1.15
+go 1.16
```
Comment (haircommander) [inline go.mod:1]: embarassingly, it's been so long that we actually don't build with anything less than 1.17 now. could you update it @rockmenjack ? also, thanks for your patience here

**Comment by haircommander** [comment]
/retest-required

LGTM @cri-o/cri-o-maintainers PTAL

**Comment by haircommander** [comment]
/hold

to figure out openshift failures

**Comment by haircommander** [comment]
sorry @rockmenjack one more merge conflict. Once you fix, and we sanity check the PR against most tests, I'll force merge. Thank you again for your patience

**Comment by haircommander** [comment]
(also feel free to give up on this PR, the CRI-O maintainer team could take it over if it's too much trouble)

**Comment by rockmenjack** [comment]
> (also feel free to give up on this PR, the CRI-O maintainer team could take it over if it's too much trouble)

yeah, looks like it is better to hand it over to maintainer team

**Comment by haircommander** [comment]
thank you for your attempt! I really appreciate your patience

**Pull request closed without merging**